content/en/blog/_posts/2021-03-24-Kubernetes-Multi-Tenancy-Usage-Models.md

```diff
@@ -0,0 +1,81 @@
+
+We will also be presenting on this content and discussing different use cases at our Kubecon EU 2021 panel session, [Multi-tenancy vs. Multi-cluster: When Should you Use What?](https://sched.co/iE66).
+
+## Multi-tenancy security considerations
+
+Cloud native security involves different system layers and lifecycle phases as described in the [Cloud Native Security Whitepaper](https://www.cncf.io/blog/2020/11/18/announcing-the-cloud-native-security-white-paper/) from CNCF SIG Security.  Without proper security measures implemented across all layers and phases, Kubernetes multi-tenancy can be compromised. 
+
+It is important for any new user to Kubernetes to realize that the default installation of a new upstream Kubernetes cluster is not secure, and you are going to need to invest in hardening it in order to avoid security issues.
+
+At a minimum, the following security measures are required:
+* Scan container images for vulnerabilities.
+* Apply [CIS Benchmarks for Kubernetes](https://www.cisecurity.org/benchmark/kubernetes/)
+* Restrict container image sources to trusted registries.
+* Implement granular [RBAC](/docs/reference/access-authn-authz/rbac/).
+* Apply default “deny-all” [network policies](https://kubernetes.io/docs/concepts/services-networking/network-policies/).
+* Implement [Kubernetes Pod Security Standards](https://kubernetes.io/docs/concepts/security/pod-security-standards/).
+
+Additional recommendations include:
+* Use a policy engine to validate configurations 
+* Implement runtime scanning
+* For stronger isolation, consider VM-based container sandboxing 
+
+While proper security is required independently of tenancy models, not having proper security controls like pod security in a shared cluster provides attackers with means to compromise tenancy models and access sensitive information across tenants increasing the overall risk profile. 
+
+
+## Multi-tenancy usage models
+
+Kubernetes clusters contain control-plane and worker nodes, with multiple software components on each. Different multi-tenancy usage models describe how these resources are shared across users.  A tenant is defined as a group of users whose workloads are isolated from users that do not belong to the tenant. There are two primary multi-tenancy usage models described below.
+
+### Namespaces-as-a-Service
+
+With the *namespaces-as-a-service* model, tenants share a cluster and tenant workloads are restricted to a set of Namespaces assigned to the tenant. The cluster control plane resources like the API server and scheduler, and worker node resources like CPU, memory, etc. are available for use across all tenants. 
+
+To isolate tenant workloads, each namespace must also contain:
+* **role bindings:** for controlling access to the namespace
+* **network policies:** to prevent network traffic across tenants
+* **resource quotas:** to limit usage and ensure fairness across tenants
```

Comment:
- These could be hyperlinks.
- Isolation between tenants typically needs an extra control: SecurityContextDeny or other restriction on Pod admission (so you can't run a privileged Pod and bypass controls); possibly also PriorityClass to protect important cluster-wide services, and EventRateLimit admission with FlowSchemas.

Reply:
@stim - thanks! Added the links.

Re: pod security and admission controls, we wanted to emphasize that these are essential independently of multi-tenancy models and hence discussed this in the "Multi-tenancy security considerations" section and a added a call-out for potential multi-tenancy risks. Does that need more emphasis?

Reply:
@sftim - I believe the restructuring helps address this. Let me know if you recommend additional changes.